Fixes #36650 - change the OS default hash function

ares opened the pull request:
The default hash function should be more secure SHA512. NSA has recommended SHA512 since RHEL 5, so it's likely widely adopted. SHA256 is no longer considered supported and is prohibitted by some policies.


<!---

Thank you for contributing to The Foreman, please read the
[following guide](https://www.theforeman.org/contribute.html), in short:

* [Create an issue](https://projects.theforeman.org/projects/foreman/issues)
* Reference the issue via `Fixes #1234` in the commit message
* Prefer present-tense, imperative-style commit messages
* Mark all strings for translation, see [1]
* Suggest prerequisites for testing and testing scenarios following example above.
* Prepend `[WIP]` for work in progress to prevent bots from triggering actions
* Be patient, we will do our best to take a look as soon as we can
* Explain the purpose of the PR, attach screenshots if applicable
* List all prerequisites for testing (e.g. VMware cluster, two smart proxies...)
* Reviewers often use extensive list of items to check, have a look prior submitting [2]
* Be nice and respectful

1: https://projects.theforeman.org/projects/foreman/wiki/Translating#Translating-for-developers
2: https://github.com/theforeman/foreman/blob/develop/developer_docs/pr_review.asciidoc
-->

ares committed:
Fixes #36650 - change the OS default hash function

The default hash function should be more secure SHA512.
NSA has recommended SHA512 since RHEL 5, so it's likely
widely adopted. SHA256 is no longer considered supported
and is prohibitted by some policies.

ekohl reviewed (commented):
Test failures look related, but :+1: for this. I'd also consider a cherry pick to 3.11 since it's still in RC if we get this finished before Tuesday.

ares commented:
Updated. One test checking the encryption hardcoded $5 as a prefix, I changed that to $6 as we don't really change the function that often and it's the new default.

Also template snapshots had to be regenerated. If this gets merged quickly, I'm happy to show that on the demo this Thursday.

ofedoren reviewed (approved):
Thanks, @ares and @ekohl ! Katello failures don't seem to be related.

ekohl commented:
When reviewing a different PR I realized that if a host doesn't have an OS set, it will still default to SHA256.

https://github.com/theforeman/foreman/blob/f2e6bca32c5b7f1e4a97d9ced2c6ead7df14716b/app/models/concerns/host_common.rb#L174

This ends up calling `passw_crypt` without a second argument:
https://github.com/theforeman/foreman/blob/f2e6bca32c5b7f1e4a97d9ced2c6ead7df14716b/app/services/password_crypt.rb#L15

So that still defaults to SHA256.

In hindsight, we should have really defined a constant `DEFAULT_HASH_ALGORHITHM` so I opened https://github.com/theforeman/foreman/pull/10231.
